### PR TITLE
move substitute lib to open source

### PR DESCRIPTION
**Pull request by linbinyu**
Summary: as title

Test Plan: buck build //xplat/caffe2:generated_aten_config_header

Reviewed By: kit1980

Differential Revision: D36989348

### CODE REVIEW

File: tools/substitute.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
```
Comment (malfet) [inline tools/substitute.py:1]: Those looks like artifacts from Python-2 era
```suggestion
```

File: tools/substitute.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    options = parser.parse_args()
+
+    with open(options.input_file, "r") as f:
+        contents = f.read()
+
+    output_file = os.path.join(options.install_dir, options.output_file)
+    try:
+        os.makedirs(os.path.dirname(output_file))
+    except OSError:
+        pass
```
Comment (malfet) [inline tools/substitute.py:23]: Wouldn't `exists_ok` be the right solution here (again, present only in Python-3.2+)
```suggestion
     os.makedirs(os.path.dirname(output_file), exists_ok=True)
```

**Review by malfet — APPROVED** [review]
LGTM, but can we make it a bit more Python-3 friendly?

File: tools/substitute.py (hunk 2 above)
Comment (linbinyu) [reply tools/substitute.py:23]: I just copied this file. I will do what you said and see if it works.

**Pull request closed without merging**